backends-common/elasticsearch-v8/src/main/java/org/apache/james/backends/es/v8/IndexCreationFactory.java

```diff
@@ -0,0 +1,376 @@
+                }
+            }
+        }
+
+        public static Builder.RequireNbShards builder() {
+            return nbShards -> nbReplica -> waitForActiveShards -> indexName
+                -> new Builder.FinalStage(nbShards, nbReplica, waitForActiveShards, indexName);
+        }
+
+        private final int nbShards;
+        private final int nbReplica;
+        private final int waitForActiveShards;
+        private final IndexName indexName;
+        private final ImmutableList<AliasName> aliases;
+        private final ImmutableList<IndexCreationCustomElement> customAnalyzers;
+        private final ImmutableList<IndexCreationCustomElement> customTokenizers;
+
+        private IndexCreationPerformer(int nbShards, int nbReplica, int waitForActiveShards, IndexName indexName,
+                                       ImmutableList<AliasName> aliases, ImmutableList<IndexCreationCustomElement> customAnalyzers,
+                                       ImmutableList<IndexCreationCustomElement> customTokenizers) {
+            this.nbShards = nbShards;
+            this.nbReplica = nbReplica;
+            this.waitForActiveShards = waitForActiveShards;
+            this.indexName = indexName;
+            this.aliases = aliases;
+            this.customAnalyzers = customAnalyzers;
+            this.customTokenizers = customTokenizers;
+        }
+
+        public ReactorElasticSearchClient createIndexAndAliases(ReactorElasticSearchClient client,
+                                                                Optional<IndexSettings> indexSettings,
+                                                                Optional<TypeMapping> mappingContent) {
+            Preconditions.checkNotNull(indexName);
+            try {
+                createIndexIfNeeded(client, indexName, indexSettings.orElse(generateSetting()), mappingContent);
+                aliases.forEach(Throwing.<AliasName>consumer(alias -> createAliasIfNeeded(client, indexName, alias))
+                    .sneakyThrow());
+            } catch (IOException e) {
+                LOGGER.error("Error while creating index : ", e);
+            }
+            return client;
+        }
+
+        private void createAliasIfNeeded(ReactorElasticSearchClient client, IndexName indexName, AliasName aliasName) throws IOException {
+            if (!aliasExist(client, aliasName)) {
+                client.updateAliases(
+                    new UpdateAliasesRequest.Builder()
+                        .actions(new Action.Builder()
+                            .add(new AddAction.Builder()
+                                .index(indexName.getValue())
+                                .alias(aliasName.getValue())
+                                .build())
+                            .build())
+                        .build());
+            }
+        }
+
+        private boolean aliasExist(ReactorElasticSearchClient client, AliasName aliasName) {
```

Comment:
Should we merge `aliasExist` and `updateAliases` in a single pipeline?

Reply:
Let's avid too much refactorings IMO

Reply:
Could be a refactoring in an other PR, too much changes already as stated by @chibenwa